src/main/java/org/gridsuite/securityanalysis/server/SecurityAnalysisController.java

```diff
@@ -107,25 +110,30 @@ public ResponseEntity<PreContingencyResult> getNResult(@Parameter(description =
             : ResponseEntity.notFound().build();
     }
 
-    @GetMapping(value = "/results/{resultUuid}/nmk-contingencies-result", produces = APPLICATION_JSON_VALUE)
+    @GetMapping(value = "/results/{resultUuid}/nmk-contingencies-result/paged", produces = APPLICATION_JSON_VALUE)
     @Operation(summary = "Get a security analysis result from the database - NMK contingencies result")
     @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The security analysis result"),
         @ApiResponse(responseCode = "404", description = "Security analysis result has not been found")})
-    public ResponseEntity<List<ContingencyResultDTO>> getNmKContingenciesResult(@Parameter(description = "Result UUID") @PathVariable("resultUuid") UUID resultUuid) {
-        List<ContingencyResultDTO> result = service.getNmKContingenciesResult(resultUuid);
+    public ResponseEntity<Page<ContingencyResultDTO>> getNmKContingenciesResult(@Parameter(description = "Result UUID") @PathVariable("resultUuid") UUID resultUuid,
+                                                                                    @Parameter(description = "Filters") @RequestParam(name = "filters", required = false) String stringFilters,
+                                                                                    @Parameter(description = "Pagination parameters") Pageable pageable) throws JsonProcessingException {
+        List<ResourceFilterDTO> filters = ResourceFilterDTO.fromStringToList(stringFilters);
```

Comment:
can you handle the JsonProcessingException in the ResourceFilterDTO.fromStringToList and throw a custom exception to say that the provided filter format is invalid

Reply:
I would do the opposite, renaming FilterColumn into Column, because we already are in the ResourceFilter DTO so we knows it's column for filter, but I guess it's just personal preference. It is done differently in other MS ?

Reply:
fixed